--- /usr/share/perl5/Apache/ASP/StateManager.pm	2016-11-14 22:47:59.812733840 +0000
+++ /usr/share/perl5/Apache/ASP/StateManager.pm.new	2016-11-14 22:47:33.561000465 +0000
@@ -82,7 +82,7 @@
 	$self->{cookie_path}       = &config($self, 'CookiePath', undef, '/');
 	$self->{cookie_domain}     = &config($self, 'CookieDomain');
 	$self->{paranoid_session}  = &config($self, 'ParanoidSession');
-	$self->{remote_ip}         = $r->connection()->remote_ip();
+	$self->{remote_ip}         = (defined $r->useragent_ip() ? $r->useragent_ip() : $r->connection()->remote_ip());
 	$self->{session_count}     = &config($self, 'SessionCount');
 	
 	# cookieless session support, cascading values
