--- /usr/share/perl5/Apache/ASP/StateManager.pm	2011-08-15 16:25:30.000000000 +0000
+++ StateManager.pm	2016-08-09 21:15:14.772373112 +0000
@@ -439,7 +439,7 @@
     if(defined $id) {
 	unless($self->{session_url_force}) {
 	    # don't set the cookie when we are just using SessionQuery* configs
-	    my $secure = $self->{secure_session} ? '; secure' : '';
+	    my $secure = $self->{secure_session} ? '; secure; HttpOnly;' : '';
 	    my $domain = $self->{cookie_domain}  ? '; domain='.$self->{cookie_domain} : '';
 	    $self->{r}->err_headers_out->add('Set-Cookie', "$SessionCookieName=$id; path=$self->{cookie_path}".$domain.$secure);
 	}
